### PR TITLE
Add: fork observing mechanism

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
This PR extends the process module and allows to manage forked childs as well sets a limit to the number of forks possible. It also cleans up child processes.

**Why**:
In order to use IPC some sort of fork observing mechanism is necessary

**How**:

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: src/processes.c [hunk 1]
```diff
@@ -46,76 +46,224 @@
...
 
+typedef struct proc_entry
+{
+  pid_t pid;
+  int terminated;
+} proc_entry;
+
+static proc_entry *procs = NULL;
+static int initialized = 0;
+static int last;
```
Comment (nichtsfrei) [inline src/processes.c:57]: Considering that it is a relatively small array it may be better to find the last index when we need it instead of storing it in a global variable?

File: src/processes.h [hunk 2]
```diff
@@ -28,9 +28,25 @@
...
 
+#define FORKFAILED -1
+#define NOINIT -2
+#define PROCSFULL -3
+#define NOCHILD -4
+
 typedef void (*process_func_t) (void *);
+
+void
+init_procs (int max);
```
Comment (nichtsfrei) [inline src/processes.h:39]: maybe it makes sense to apply a prefix to the methods defined here? Like procs_terminate_child, procs_init, ... to make it easier to find the implementation of those methods when looking for them.

**Review by nichtsfrei — CHANGES_REQUESTED** [review]

File: src/processes.h (hunk 2 above)
Comment (Kraemii) [reply src/processes.h]: I did it for `terminate_childs` and `procs_init` as they currently are not used in the rest of openvas. Feel free to change their names too, when applying IPC

File: src/processes.c (hunk 1 above)
Comment (Kraemii) [reply src/processes.c]: Changed it

**Review by nichtsfrei — APPROVED** [review]

File: src/processes.c [hunk 3]
```diff
@@ -46,76 +46,220 @@
...
+ * @return pid of spawned process on success or one of the following errors:
+ * FORKFAILED, NOINIT, PROCSFULL
  */
 pid_t
-create_process (process_func_t function, void *argument)
+create_process (process_func_t func, void *args)
 {
-  int pid;
-
-  gvm_log_lock ();
```
Comment (jjnicola) [inline src/processes.c:104]: Why did you remove this? This code was introduced some time ago to avoid deadlock. See PR #491
